parsl-advanced-features.ipynb

```diff
@@ -469,12 +469,131 @@
     "parsl.clear()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Monitoring\n",
+    "\n",
+    "Parsl can be configured to capture fine grain monitoring information about workflows and resource usage. To enable monitoring you must add the monitoring hub to the configuration.\n",
+    "\n",
+    "Note: in this example we set the resource monitoring interval to 3 seconds so that we can capture resource information from short running tasks. In practice you will likely use a longer interval."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "import parsl\n",
+    "from parsl.monitoring.monitoring import MonitoringHub\n",
+    "from parsl.config import Config\n",
+    "from parsl.executors import HighThroughputExecutor\n",
+    "from parsl.addresses import address_by_hostname\n",
+    "\n",
+    "import logging\n",
+    "\n",
+    "config = Config(\n",
+    "   executors=[\n",
+    "       HighThroughputExecutor(\n",
+    "           label=\"local_htex\",\n",
+    "           address=address_by_hostname(),\n",
```

Comment:
Adding this because default `LocalProvider` has 4 blocks on a single node, and this would cause the non-unique primary key problem for the `node` table in monitoring

```suggestion
    "           address=address_by_hostname(),\n",
    "           provider=LocalProvider(\n",
    "               init_blocks=1,\n",
    "               max_blocks=1,\n",
    "           )\n",
```

Reply:
@ZhuozhaoLi that's a parsl bug?

Reply:
> @ZhuozhaoLi that's a parsl bug?

I think it is. Our node table has three primary key `hostname, run_id, reg_time`. If we have 4 blocks on the same node and they start very quick, I assume they will have the same primary keys. But maybe we can make the assumption here that we should not have 4 blocks on LocalProvider on one node?